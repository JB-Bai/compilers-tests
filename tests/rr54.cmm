float roQlUPOB2v(){
	if(0x58c2) Tcg4mD(TRAUy6Z);
	else return 0x40D;
	OjLNO(0x0a);
	if(ZMLrZKFJH8 <= ! GZXGh8GB / 4682) 33.729;
	else 221 / - gzHlaOUPYw;
	(YwdnbG);
	AKucKq / - O0WpKJ0Mi - OJV3bS4h6Nq && lWdHFzOSB1;
}
float;
struct{
}
Tb9sX1rdp3Z(){
	while(0047317) if(50.676) return 0x067D;
	else return ! 045124 || 036611 + Ly6eouUvDk + - ! - sCa2zEm * 84324 - 3.67E-7 + bchI4QMSqU - - FdXMN7 <= 999 + WPj7zWwZ - nAO3U()[0x7][0605] || 9889[BhEpKyfO7 && 08.285E9] * 7029 && Pt9nIxs * 752.16 = IKdQ4r0g <= roQkb5E - 0x32Ca + jeacILA_() = 044.490.dQmU + 0xebB = 0xaAB8 || 0xba - 0xb73ca - YJEwHtV && 748 <= 86.25E-4 / 278.669[! 07066 / 803.33] = MvnqBu(opUEU5t() = luh3QR0q(- 00362, ajuuPa, lLILG || 9607.54e-6 * LQXTYLjQ).FQCy7TR, 29.4, NgnWh07lv[0125620]).zvywNV.zCiXEEmHF;
	65.596e+4;
	if(066565) while(3559.32) return 020 && etotq[72] || vLcEkP(yjY6Zc);
	hNYNL6_1;
	0xDc * Lpi9n7b = 157.651e0 * C0curIsvx6W;
	838.7;
}

