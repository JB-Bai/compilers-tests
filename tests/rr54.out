Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: roQlUPOB2v
        LP
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            IF
            LP
            Exp (2)
              INT: 22722
            RP
            Stmt (2)
              Exp (2)
                ID: Tcg4mD
                LP
                Args (2)
                  Exp (2)
                    ID: TRAUy6Z
                RP
              SEMI
            ELSE
            Stmt (3)
              RETURN
              Exp (3)
                INT: 1037
              SEMI
          StmtList (4)
            Stmt (4)
              Exp (4)
                ID: OjLNO
                LP
                Args (4)
                  Exp (4)
                    INT: 10
                RP
              SEMI
            StmtList (5)
              Stmt (5)
                IF
                LP
                Exp (5)
                  Exp (5)
                    ID: ZMLrZKFJH8
                  RELOP
                  Exp (5)
                    Exp (5)
                      NOT
                      Exp (5)
                        ID: GZXGh8GB
                    DIV
                    Exp (5)
                      INT: 4682
                RP
                Stmt (5)
                  Exp (5)
                    FLOAT: 33.729000
                  SEMI
                ELSE
                Stmt (6)
                  Exp (6)
                    Exp (6)
                      INT: 221
                    DIV
                    Exp (6)
                      MINUS
                      Exp (6)
                        ID: gzHlaOUPYw
                  SEMI
              StmtList (7)
                Stmt (7)
                  Exp (7)
                    LP
                    Exp (7)
                      ID: YwdnbG
                    RP
                  SEMI
                StmtList (8)
                  Stmt (8)
                    Exp (8)
                      Exp (8)
                        Exp (8)
                          Exp (8)
                            ID: AKucKq
                          DIV
                          Exp (8)
                            MINUS
                            Exp (8)
                              ID: O0WpKJ0Mi
                        MINUS
                        Exp (8)
                          ID: OJV3bS4h6Nq
                      AND
                      Exp (8)
                        ID: lWdHFzOSB1
                    SEMI
        RC
    ExtDefList (10)
      ExtDef (10)
        Specifier (10)
          TYPE: float
        SEMI
      ExtDefList (11)
        ExtDef (11)
          Specifier (11)
            StructSpecifier (11)
              STRUCT
              LC
              RC
          FunDec (13)
            ID: Tb9sX1rdp3Z
            LP
            RP
          CompSt (13)
            LC
            StmtList (14)
              Stmt (14)
                WHILE
                LP
                Exp (14)
                  INT: 20175
                RP
                Stmt (14)
                  IF
                  LP
                  Exp (14)
                    FLOAT: 50.675999
                  RP
                  Stmt (14)
                    RETURN
                    Exp (14)
                      INT: 1661
                    SEMI
                  ELSE
                  Stmt (15)
                    RETURN
                    Exp (15)
                      Exp (15)
                        Exp (15)
                          Exp (15)
                            NOT
                            Exp (15)
                              INT: 19028
                          OR
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                Exp (15)
                                  Exp (15)
                                    Exp (15)
                                      Exp (15)
                                        INT: 15753
                                      PLUS
                                      Exp (15)
                                        ID: Ly6eouUvDk
                                    PLUS
                                    Exp (15)
                                      Exp (15)
                                        MINUS
                                        Exp (15)
                                          NOT
                                          Exp (15)
                                            MINUS
                                            Exp (15)
                                              ID: sCa2zEm
                                      STAR
                                      Exp (15)
                                        INT: 84324
                                  MINUS
                                  Exp (15)
                                    FLOAT: 0.000000
                                PLUS
                                Exp (15)
                                  ID: bchI4QMSqU
                              MINUS
                              Exp (15)
                                MINUS
                                Exp (15)
                                  ID: FdXMN7
                            RELOP
                            Exp (15)
                              Exp (15)
                                Exp (15)
                                  INT: 999
                                PLUS
                                Exp (15)
                                  ID: WPj7zWwZ
                              MINUS
                              Exp (15)
                                Exp (15)
                                  Exp (15)
                                    ID: nAO3U
                                    LP
                                    RP
                                  LB
                                  Exp (15)
                                    INT: 7
                                  RB
                                LB
                                Exp (15)
                                  INT: 389
                                RB
                        OR
                        Exp (15)
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                INT: 9889
                              LB
                              Exp (15)
                                Exp (15)
                                  ID: BhEpKyfO7
                                AND
                                Exp (15)
                                  FLOAT: 8285000192.000000
                              RB
                            STAR
                            Exp (15)
                              INT: 7029
                          AND
                          Exp (15)
                            Exp (15)
                              ID: Pt9nIxs
                            STAR
                            Exp (15)
                              FLOAT: 752.159973
                      ASSIGNOP
                      Exp (15)
                        Exp (15)
                          Exp (15)
                            ID: IKdQ4r0g
                          RELOP
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                ID: roQkb5E
                              MINUS
                              Exp (15)
                                INT: 13002
                            PLUS
                            Exp (15)
                              ID: jeacILA_
                              LP
                              RP
                        ASSIGNOP
                        Exp (15)
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                FLOAT: 44.490002
                              DOT
                              ID: dQmU
                            PLUS
                            Exp (15)
                              INT: 3771
                          ASSIGNOP
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                INT: 43704
                              OR
                              Exp (15)
                                Exp (15)
                                  Exp (15)
                                    Exp (15)
                                      INT: 186
                                    MINUS
                                    Exp (15)
                                      INT: 750538
                                  MINUS
                                  Exp (15)
                                    ID: YJEwHtV
                                AND
                                Exp (15)
                                  Exp (15)
                                    INT: 748
                                  RELOP
                                  Exp (15)
                                    Exp (15)
                                      FLOAT: 0.008625
                                    DIV
                                    Exp (15)
                                      Exp (15)
                                        FLOAT: 278.669006
                                      LB
                                      Exp (15)
                                        Exp (15)
                                          NOT
                                          Exp (15)
                                            INT: 3638
                                        DIV
                                        Exp (15)
                                          FLOAT: 803.330017
                                      RB
                            ASSIGNOP
                            Exp (15)
                              Exp (15)
                                Exp (15)
                                  ID: MvnqBu
                                  LP
                                  Args (15)
                                    Exp (15)
                                      Exp (15)
                                        ID: opUEU5t
                                        LP
                                        RP
                                      ASSIGNOP
                                      Exp (15)
                                        Exp (15)
                                          ID: luh3QR0q
                                          LP
                                          Args (15)
                                            Exp (15)
                                              MINUS
                                              Exp (15)
                                                INT: 242
                                            COMMA
                                            Args (15)
                                              Exp (15)
                                                ID: ajuuPa
                                              COMMA
                                              Args (15)
                                                Exp (15)
                                                  Exp (15)
                                                    ID: lLILG
                                                  OR
                                                  Exp (15)
                                                    Exp (15)
                                                      FLOAT: 0.009608
                                                    STAR
                                                    Exp (15)
                                                      ID: LQXTYLjQ
                                          RP
                                        DOT
                                        ID: FQCy7TR
                                    COMMA
                                    Args (15)
                                      Exp (15)
                                        FLOAT: 29.400000
                                      COMMA
                                      Args (15)
                                        Exp (15)
                                          Exp (15)
                                            ID: NgnWh07lv
                                          LB
                                          Exp (15)
                                            INT: 43920
                                          RB
                                  RP
                                DOT
                                ID: zvywNV
                              DOT
                              ID: zCiXEEmHF
                    SEMI
              StmtList (16)
                Stmt (16)
                  Exp (16)
                    FLOAT: 655960.000000
                  SEMI
                StmtList (17)
                  Stmt (17)
                    IF
                    LP
                    Exp (17)
                      INT: 28021
                    RP
                    Stmt (17)
                      WHILE
                      LP
                      Exp (17)
                        FLOAT: 3559.320068
                      RP
                      Stmt (17)
                        RETURN
                        Exp (17)
                          Exp (17)
                            Exp (17)
                              INT: 16
                            AND
                            Exp (17)
                              Exp (17)
                                ID: etotq
                              LB
                              Exp (17)
                                INT: 72
                              RB
                          OR
                          Exp (17)
                            ID: vLcEkP
                            LP
                            Args (17)
                              Exp (17)
                                ID: yjY6Zc
                            RP
                        SEMI
                  StmtList (18)
                    Stmt (18)
                      Exp (18)
                        ID: hNYNL6_1
                      SEMI
                    StmtList (19)
                      Stmt (19)
                        Exp (19)
                          Exp (19)
                            Exp (19)
                              INT: 220
                            STAR
                            Exp (19)
                              ID: Lpi9n7b
                          ASSIGNOP
                          Exp (19)
                            Exp (19)
                              FLOAT: 157.651001
                            STAR
                            Exp (19)
                              ID: C0curIsvx6W
                        SEMI
                      StmtList (20)
                        Stmt (20)
                          Exp (20)
                            FLOAT: 838.700012
                          SEMI
            RC
